client/checkout/classic/3ds-flow-handling.js

```diff
@@ -22,37 +22,27 @@ const cleanupURL = () => {
 };
 
 export const showAuthenticationModalIfRequired = ( api ) => {
-	const url = window.location.href;
 	const paymentMethodId = document.querySelector( '#wcpay-payment-method' )
 		?.value;
 
 	const confirmation = api.confirmIntent(
-		url,
+		window.location.href,
 		shouldSavePaymentPaymentMethod() ? paymentMethodId : null
 	);
 
 	// Boolean `true` means that there is nothing to confirm.
 	if ( confirmation === true ) {
-		return;
```

Comment:
I changed the function to return a promise, so that the caller could block/unblock the UI